vulture_os/toolkit/api_parser/perception_point_x_ray/perception_point_x_ray.py

```diff
@@ -0,0 +1,237 @@
+#!/home/vlt-os/env/bin/python
+"""This file is part of Vulture OS.
+
+Vulture OS is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Vulture OS is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Vulture OS.  If not, see http://www.gnu.org/licenses/.
+"""
+__author__ = "nlancon"
+__credits__ = []
+__license__ = "GPLv3"
+__version__ = "4.0.0"
+__maintainer__ = "Vulture OS"
+__email__ = "contact@vultureproject.org"
+__doc__ = 'PERCEPTION POINT X RAY API Parser'
+__parser__ = 'PERCEPTION_POINT_X_RAY'
+
+import logging
+from django.conf import settings
+from django.utils import timezone
+
+from toolkit.api_parser.api_parser import ApiParser
+
+import requests
+import json
+
+from datetime import datetime, timedelta
+
+logging.config.dictConfig(settings.LOG_SETTINGS)
+logger = logging.getLogger('api_parser')
+
+
+class PerceptionPointXRayAPIError(Exception):
+    pass
+
+
+class PerceptionPointXRayParser(ApiParser):
+
+    HEADERS = {
+        "Accept": "application/json"
+    }
+
+    LOG_TYPES = ["SUS", "MAL", "SPM"]  # Malicious, spam and suspicious
```

Comment:
Won't we ever have to select which type(s) of log to fetch?

Reply:
BT wants to fetch all logs